library/src/main/java/com/datatorrent/lib/io/block/ReaderContext.java

```diff
@@ -190,7 +190,7 @@ protected Entity readEntity() throws IOException
           if (bytesRead == -1) {
             break;
           }
-          strBuffer = new String(buffer);
+          strBuffer = new String(buffer,0, bytesRead);
```

Comment:
strBuffer is misnomer. It refers string and not string buffer.. Can you please refactor this too